dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java

```diff
@@ -844,7 +862,7 @@ public Response toolsIdVersionsVersionIdTypeFilesGet(String id, DescriptorType t
                     // Matching the workflow path in a workflow automatically indicates that the file is a primary descriptor
                     primaryDescriptorPaths.add(workflowVersion.getWorkflowPath());
                     Set<SourceFile> sourceFiles = workflowVersion.getSourceFiles();
-                    if ("zip".equalsIgnoreCase(format)) {
+                    if ("zip".equalsIgnoreCase(format) || consumesHeaderZipOnly) {
                         return getZipResponse(sourceFiles, workflow.getWorkflowPath(), workflowVersion.getName(), Paths.get(workflowVersion.getWorkingDirectory()));
```

Comment:
Is the Content-type header of the response set correctly, if accept is `application/json` and `format=zip`?

Reply:
There should be a HTTP 400 from line 83 if only json is accepted but the user specified zip

Reply:
In hindsight, maybe the TRS spec should have allowed this via content-negotiation only, but I think you said there are issues with Swagger and content negotiation. And it's in the spec already anyway, so this is good.